hkdf/src/hkdf.rs

```diff
@@ -27,19 +27,23 @@ impl<D> Hkdf<D>
           D::OutputSize: ArrayLength<u8>
 {
     #[deprecated(since="0.4.0", note="renamed to extract, with arguments in reversed order")]
-    pub fn new(ikm: &[u8], salt: &[u8]) -> Hkdf<D> {
+    pub fn new(ikm: &[u8], salt: Option<&[u8]>) -> Hkdf<D> {
         Self::extract(salt, ikm)
     }
 
     /// The RFC5869 HKDF-Extract operation
-    pub fn extract(salt: &[u8], ikm: &[u8]) -> Hkdf<D> {
+    pub fn extract(salt: Option<&[u8]>, ikm: &[u8]) -> Hkdf<D> {
         // The hmac-0.5 MAC trait (which provides new()) is now defined to
         // return a Result, apparently to support things like CMAC which
         // require a specific key length. As far as I can tell, HMAC in
         // particular can only return an Ok(), since HMAC accepts any length
         // of bytes as its key. So we use unwrap() here, rather than exposing
         // the error to our caller. This might change in a future version.
-        let mut hmac = Hmac::<D>::new(salt).unwrap();
+        let mut hmac = match salt {
+            Some(s) => Hmac::<D>::new(s),
+            None => Hmac::<D>::new(&generic_array::GenericArray::<u8, D::OutputSize>::default()),
+        }.expect("HMAC can accept keys of any size");
```

Comment:
Shouldn't this be readable as an error message?

Reply:
`Hmac::new(key)` never returns an error, so message in the `expect`will never be used. It's just a message for those who will read the code.